<a target="_blank" rel="noopener noreferrer" href="https://colab.research.google.com/github/ccaudek/ds4psy_2023/blob/main/125_expval_var.ipynb">![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)</a>

(random_variables_notebook)=
# Variabili casuali 

In [1]:
source("../_common.R")
set.seed(42)

## Variabili Casuali: Natura, Tipologie e Distribuzione

### Natura delle Variabili Casuali

Una variabile casuale è un costrutto matematico fondamentale nella teoria della probabilità e nelle applicazioni statistiche. Formalmente, è definita come una funzione che mappa uno spazio campione $ S $ a un sottoinsieme dei numeri reali $ \mathbb{R} $. In termini più intuitivi, una variabile casuale quantifica gli esiti di un fenomeno aleatorio, assegnando un valore numerico a ciascuno di essi. 

$$
\text{Variabile Casuale (v.c.)} : S \rightarrow \mathbb{R}
$$

### Classificazione delle Variabili Casuali

Le variabili casuali possono essere categorizzate in due classi principali:

1. **Variabili Casuali Discrete**: Queste variabili assumono un insieme finito o numerabile di valori distinti. Ad esempio, la frequenza con cui un individuo controlla il proprio smartphone in un'ora è una variabile casuale discreta.

2. **Variabili Casuali Continue**: Queste variabili possono assumere un'infinità di valori all'interno di un determinato intervallo continuo. Un esempio potrebbe essere il tempo di reazione di un individuo a un particolare stimolo visuale.

### Variabile Casuale come Entità Latente

Una variabile casuale può anche essere concepita come una "variabile latente": un parametro intrinseco ma non direttamente osservabile. Questa variabile latente rappresenta l'intero spettro di esiti potenzialmente osservabili in un esperimento, ma rimane in uno stato di "latenza" fino a quando l'esperimento non viene effettuato. Al momento della realizzazione dell'esperimento, la variabile latente si "attualizza" in un esito specifico, che viene quindi registrato come dato empirico. Ad esempio, in un test psicometrico come un test di IQ, l'intelligenza è la variabile latente che si manifesta attraverso i punteggi ottenuti dai partecipanti al test.

### Notazione

È convenzionale denotare le variabili casuali con lettere maiuscole (ad esempio, $X, Y, Z$) e i loro valori realizzati (esiti osservati) con lettere minuscole (ad esempio, $x, y, z$).

### Distribuzione di una Variabile Casuale

Una volta introdotta la variabile casuale, si pone la questione di come descrivere l'insieme di tutti i suoi esiti possibili e le relative probabilità. Questo ci introduce al concetto di "distribuzione" di una variabile casuale.

#### Che Cos'è una Distribuzione?

Una distribuzione di una variabile casuale fornisce una descrizione dettagliata degli esiti possibili e delle loro probabilità associate. Per esempio, nel lancio di un dado standard a sei facce, ogni faccia ha una probabilità di $ \frac{1}{6} $ di apparire. In questo caso, la distribuzione della variabile casuale sarà un elenco che collega ogni esito numerico (da 1 a 6) alla sua probabilità di occorrenza ($ \frac{1}{6} $).

Questo quadro distributivo è cruciale per comprendere il comportamento delle variabili casuali, consentendo analisi inferenziali e previsionali più accurate.

In sintesi, la variabile casuale agisce come un modello astratto che esprime la totalità degli esiti possibili in un fenomeno aleatorio. Solo quando l'esperimento viene condotto, la variabile casuale esce dal suo stato latente per manifestarsi come un esito osservabile e specifico, che può quindi essere analizzato statisticamente.

## Valore atteso

Spesso è utile sintetizzare la distribuzione di una variabile casuale tramite indicatori caratteristici.  Questi indicatori permettono di cogliere le caratteristiche principali della distribuzione, come la posizione (cioè il baricentro) e la variabilità (cioè la dispersione attorno ad un centro). In questo modo, si può avere una descrizione sintetica della distribuzione di probabilità della variabile casuale. In questo capitolo introdurremo i concetti di valore atteso e di varianza di una variabile casule. 

Quando vogliamo conoscere il comportamento tipico di una variabile casuale spesso vogliamo sapere qual è il suo "valore tipico". La nozione di "valore tipico", tuttavia, è ambigua. Infatti, essa può essere definita in almeno tre modi diversi:

-   la *media* (somma dei valori divisa per il numero dei valori),
-   la *mediana* (il valore centrale della distribuzione, quando la variabile è ordinata in senso crescente o decrescente),
-   la *moda* (il valore che ricorre più spesso).

Per esempio, la media di $\{3, 1, 4, 1, 5\}$ è $\frac{3+1+4+1+5}{5} = 2.8$, la mediana è $3$ e la moda è $1$. Tuttavia, la teoria delle probabilità si occupa di variabili casuali piuttosto che di sequenze di numeri. Diventa dunque necessario precisare che cosa intendiamo per "valore tipico" quando facciamo riferimento alle variabili casuali. Giungiamo così alla seguente definizione.

```{admonition} Definizione
Sia $Y$ è una variabile casuale discreta che assume i valori $y_1, \dots, y_n$ con distribuzione $P(Y = y_i) = p(y_i)$. Per definizione il *valore atteso* di $Y$, $\mathbb{E}(Y)$, è

$$
\mathbb{E}(Y) = \sum_{i=1}^n y_i \cdot p(y_i).
$$ (eq-expval-discr)
```

A parole: il valore atteso (o speranza matematica, o aspettazione, o valor medio) di una variabile casuale è definito come la somma di tutti i valori che la variabile casuale può prendere, ciascuno pesato dalla probabilità con cui il valore è preso.

````{prf:example}
:label: rv-expl-2
Calcoliamo il valore atteso della variabile casuale $Y$ corrispondente al lancio di una moneta equilibrata (testa: *Y* = 1; croce: *Y* = 0).
````

$$
\mathbb{E}(Y) = \sum_{i=1}^{2} y_i \cdot P(y_i) = 0 \cdot \frac{1}{5} + 1 \cdot \frac{1}{5} = 0.5.
$$

````{prf:example}
:label: rv-expl-3
Calcoliamo il valore atteso della variabile casuale $X$ corrispondente alla somma dei punti ottenuti dal lancio di due dadi equilibrati a sei facce. 
````

Abbiamo visto nel capitolo {ref}`intro_prob_notebook` che $X$ può assumere i valori [2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12] con distribuzione di massa di probabilità pari a [1/36, 2/36, 3/36, 4/36, 5/36, 6/36, 5/36, 4/36, 3/36, 2/36, 1/36]. Applicando l'eq. {eq}`eq-expval-discr` otteniamo:

$$
\mathbb{E}(X) = \sum_{i=1}^{11} x_i \cdot P(x_i) = 2 \cdot \frac{1}{36} + 3 \cdot \frac{2}{36} + \dots + 12 \cdot \frac{1}{36} = 7.0.
$$

Svolgiamo ora l'esercizio in `R`. 

Definisco i valori della variabile casuale $X$ e li trasformiamo in un array NumPy:

In [6]:
x <- 2:12
x

[1]  2  3  4  5  6  7  8  9 10 11 12

In [6]:
x = np.array(list(range(2, 13)))
x

array([ 2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12])

Per trovare la distribuzione di massa della variabile $X$ ripeto qui il codice che abbiamo usato nel capitolo {ref}`intro_prob_notebook`.

In [8]:
r <- 1:6
sample <- expand.grid(r, r)

px <- numeric(11)

for (i in 2:12) {
    event <- subset(sample, Var1 + Var2 == i)
    px[i - 1] <- nrow(event) / nrow(sample)
}

print(px)


 [1] 0.02777778 0.05555556 0.08333333 0.11111111 0.13888889 0.16666667
 [7] 0.13888889 0.11111111 0.08333333 0.05555556 0.02777778


Calcolo ora il valore atteso della $X$ usando l'eq. {eq}`eq-expval-discr`:

In [11]:
ex <- sum(x * px)
print(ex)

[1] 7


In alternativa, posso usare le funzioni del modulo `rv_discrete` della libreria `stats`:

### Interpretazione

Il valore atteso corrisponde alla media aritmetica di un grande numero di realizzazioni indipendenti della variabile casuale.

Per fare un esempio, ritorniamo all'esempio precedente relativo al lancio di due dadi bilanciati a sei facce nel quale $X$ rappresenta la "somma dei due dadi". Per interpretare il valore atteso, simuliamo un grande numero di realizzazioni indipendenti della $X$ mediante la funzione `random.choice()` della libreria NumPy. Tale funzione prende come argomenti i valori della variabile casuale, il numero di ripetizioni indipedenti (qui 1,000,000) e la distribuzione di massa di probabilità:

In [12]:
x_samples <- sample(x, size = 1000000, replace = TRUE, prob = px)

L'istruzione `sample(x, size = 1000000, replace = TRUE, prob = px)` genera un array di 1.000.000 di elementi (parametro `size`), scelti casualmente dall'array `x` con le probabilità specificate nell'array `px`. In particolare, `x` è l'array di cui si vuole effettuare una scelta casuale e `px` è un array che contiene le probabilità associate ad ogni elemento di `x`. 

Come ci aspettavamo, per un grande numero di realizzazioni indipendenti della $X$, la media aritmetica approssima il valore atteso:

In [13]:
mean(x_samples)

[1] 7.000703

### Proprietà del valore atteso

La proprietà più importante del valore atteso è la linearità: il valore atteso di una somma di variabili casuali è uguale alla somma dei lori rispettivi valori attesi:

$$
\mathbb{E}(X + Y) = \mathbb{E}(X) + \mathbb{E}(Y).
$$ (eq-prop-expval-linearity)

L'eq. {eq}`eq-prop-expval-linearity` sembra ragionevole quando $X$ e $Y$ sono indipendenti, ma è anche vera quando $X$ e $Y$ sono associati. Abbiamo anche che

$$
\mathbb{E}(cY) = c \mathbb{E}(Y).
$$ (eq-prop-expval-const)

L'eq. {eq}`eq-prop-expval-const` ci dice che possiamo estrarre una costante dall'operatore di valore atteso. Tale proprietà si estende a qualunque numero di variabili casuali. Infine, se due variabili casuali $X$ e $Y$ sono indipendenti, abbiamo che

$$
\mathbb{E}(X Y) = \mathbb{E}(X) \mathbb{E}(Y). 
$$ (eq-expval-prod-ind-rv)

La media aritmetica $\textstyle {\bar  {X}}={\frac  {X_{1}+\ldots +X_{n}}{n}}$ di $n$ variabili casuali indipendenti aventi la medesima distribuzione di media $\mu$ ha valore atteso 

$$
\mathbb{E}(\bar{X}) = \frac{1}{n} \mathbb{E}(X_1)+ \dots \mathbb{E}(X_n) = \frac{1}{n} n \mathbb{E}(X) = \mu.
$$

````{prf:example}
:label: rv-expl-4

Consideriamo il seguente esperimento casuale. Sia $Y$ il numero che si ottiene dal lancio di un dado equilibrato a sei facce e $Y$ il numero di teste prodotto dal lancio di una moneta equilibrata (0 oppure 1). Troviamo il valore atteso di $X+Y$.
````

Per risolvere il problema iniziamo a costruire lo spazio campione dell'esperimento casuale.

| $x /\ y$ |   1    |   2    |   3    |   4    |   5    |   6    |
|:--------------------:|:------:|:------:|:------:|:------:|:------:|:------:|
|          0           | (0, 1) | (0, 2) | (0, 3) | (0, 4) | (0, 5) | (0, 6) |
|          1           | (1, 1) | (1, 2) | (1, 3) | (1, 4) | (1, 5) | (1, 6) |

ovvero

| $x /\ y$ |  1  |  2  |  3  |  4  |  5  |  6  |
|:--------------------:|:---:|:---:|:---:|:---:|:---:|:---:|
|          0           |  1  |  2  |  3  |  4  |  5  |  6  |
|          1           |  2  |  3  |  4  |  5  |  6  |  7  |

Il risultato del lancio del dado è indipendente dal risultato del lancio della moneta. Pertanto, ciascun evento elementare dello spazio campione avrà la stessa probabilità di verificarsi, ovvero $P(\omega) = \frac{1}{12}$. Il valore atteso di $X+Y$ è dunque uguale a:

$$
\mathbb{E}(X+Y) = 1 \cdot \frac{1}{12} + 2 \cdot \frac{1}{12} + \dots + 7 \cdot \frac{1}{12} = 4.0.
$$

Si ottiene lo stesso risultato usando l'eq. {eq}`eq-prop-expval-linearity`:

$$
\mathbb{E}(X+Y) = \mathbb{E}(X) + E(Y) = 3.5 + 0.5 = 4.0.
$$

Svolgiamo ora l'esercizio in `R`.

In [20]:
coin <- 0:1
die <- 1:6

sample <- expand.grid(coin, die)
names(sample) <- c("coin", "die")
sample


coin die
1  0    1  
2  1    1  
3  0    2  
4  1    2  
5  0    3  
6  1    3  
7  0    4  
8  1    4  
9  0    5  
10 1    5  
11 0    6  
12 1    6

In [21]:
px <- numeric(7)

for (i in 1:7) {
    event <- subset(sample, coin + die == i)
    px[i] <- nrow(event) / nrow(sample)
    cat(sprintf("P(X + Y = %d) = %d / %d\n", i, nrow(event), nrow(sample)))
}


P(X + Y = 1) = 1 / 12
P(X + Y = 2) = 2 / 12
P(X + Y = 3) = 2 / 12
P(X + Y = 4) = 2 / 12
P(X + Y = 5) = 2 / 12
P(X + Y = 6) = 2 / 12
P(X + Y = 7) = 1 / 12


In [22]:
x <- 1:7
expected_value <- sum(x * px)
expected_value

[1] 4

````{prf:example}
:label: rv-expl-5

Consideriamo le variabili casuali $X$ e $Y$ definite nel caso del lancio di tre monete equilibrate, dove $X$ conta il numero delle teste nei tre lanci e $Y$ conta il numero delle teste al primo lancio. Si calcoli il valore atteso di $Z = X \cdot Y$.
````

La distribuzione di probabilità congiunta $P(X, Y)$ è fornita nella tabella seguente.

| $x /\ y$ |  0  |  1  | $p(Y)$ |
|:--------------------:|:---:|:---:|:------:|
|          0           | 1/8 |  0  |  1/8   |
|          1           | 2/8 | 1/8 |  3/8   |
|          2           | 1/8 | 2/8 |  3/8   |
|          3           |  0  | 1/8 |  1/8   |
|        $p(y)$        | 4/8 | 4/8 |  1.0   |

Il calcolo del valore atteso di $XY$ si riduce a

$$
\mathbb{E}(Z) = 1 \cdot \frac{1}{8} + 2 \cdot \frac{2}{8} + 3 \cdot \frac{1}{8} = 1.0.
$$

Si noti che le variabili casuali $Y$ e $Y$ non sono indipendenti. Dunque non possiamo usare l'eq. {eq}`eq-expval-prod-ind-rv`. Infatti, il valore atteso di $X$ è

$$
\mathbb{E}(X) = 1 \cdot \frac{3}{8} + 2 \cdot \frac{3}{8} + 3 \cdot \frac{1}{8} = 1.5
$$

e il valore atteso di $Y$ è

$$
\mathbb{E}(Y) = 0 \cdot \frac{4}{8} + 1 \cdot \frac{4}{8} = 0.5.
$$

Perciò

$$
1.5 \cdot 0.5 \neq 1.0.
$$

Svolgiamo l'esercizio in `R`.

In [24]:
# Creazione di tutte le possibili combinazioni di tre lanci di moneta
r <- 0:1
sample <- expand.grid(r, r, r)

# Calcolo delle probabilità per ogni valore di Z
for (i in 0:3) {
    event <- subset(sample, Var1 + Var2 + Var3 == i)
    cat(sprintf("P(Z = %d) : %d / %d\n", i, nrow(event), nrow(sample)))
}

P(Z = 0) : 1 / 8
P(Z = 1) : 3 / 8
P(Z = 2) : 3 / 8
P(Z = 3) : 1 / 8


In [25]:
# Calcolo del valore atteso di Z
z <- 0:3
pz <- c(4 / 8, 1 / 8, 2 / 8, 1 / 8)
expected_value <- sum(z * pz)
expected_value

[1] 1

### Variabili casuali continue

Nel caso di una variabile casuale continua $Y$ il valore atteso diventa:

$$
\mathbb{E}(Y) = \int_{-\infty}^{+\infty} y p(y) \,\operatorname{d}\!y.
$$ (eq-def-ev-rv-cont)

Anche in questo caso il valore atteso è una media ponderata della $y$, nella quale ciascun possibile valore $y$ è ponderato per il corrispondente valore della densità $p(y)$. Possiamo leggere l'integrale pensando che $y$ rappresenti l'ampiezza delle barre infinitamente strette di un istogramma, con la densità $p(y)$ che corrisponde all'altezza di tali barre e la notazione $\int_{-\infty}^{+\infty}$ che corrisponde ad una somma.[^018_expval_var-1]

[^018_expval_var-1]: Per il significato della notazione di integrale, si veda l'appendice {ref}`calculus-appendix`.

#### Moda

Un'altra misura di tendenza centrale delle variabili casuali continue è la moda. La moda di $Y$ individua il valore $y$ più plausibile, ovvero il valore $y$ che massimizza la funzione di densità $p(y)$:

$$
Mo(Y) = \text{argmax}_y p(y).
$$ (eq-def-mode)

```{note}
La notazione $\text{argmax}_y p(y)$ significa: il valore $y$ tale per cui la funzione $p(y)$ assume il suo valore massimo.
```

## Varianza

La seconda più importante proprietà di una variabile casuale, dopo che conosciamo il suo valore atteso, è la *varianza*.

```{admonition} Definizione
Se $Y$ è una variabile casuale discreta con distribuzione $p(y)$, per definizione la varianza di $Y$, $\mathbb{V}(Y)$, è

$$
\mathbb{V}(Y) = \mathbb{E}\Big[\big(Y - \mathbb{E}(Y)\big)^2\Big].
$$ (eq-def-var-rv)
```

A parole: la varianza è la deviazione media quadratica della variabile dalla sua media.[^018_expval_var-2] Se denotiamo $\mathbb{E}(Y) = \mu$, la varianza $\mathbb{V}(Y)$ diventa il valore atteso di $(Y - \mu)^2$.

[^018_expval_var-2]: Data una variabile casuale $Y$ con valore atteso $\mathbb{E}(Y)$, le "distanze" tra i valori di $Y$ e il valore atteso $\mathbb{E}(Y)$ definiscono la variabile casuale $Y - \mathbb{E}(Y)$ chiamata *scarto*, oppure *deviazione* oppure *variabile casuale centrata*. La variabile $Y - \mathbb{E}(Y)$ equivale ad una traslazione di sistema di riferimento che porta il valore atteso nell'origine degli assi. Si può dimostrare facilmente che il valore atteso della variabile scarto $Y - \mathbb{E}(Y)$ vale zero, dunque la media di tale variabile non può essere usata per quantificare la "dispersione" dei valori di $Y$ relativamente al suo valore medio. Occorre rendere sempre positivi i valori di $Y - \mathbb{E}(Y)$ e tale risultato viene ottenuto considerando la variabile casuale $\left(Y - \mathbb{E}(Y)\right)^2$.


````{prf:example}
:label: rv-expl-6

Posta $S$ uguale alla somma dei punti ottenuti nel lancio di due dadi equilibrati, si calcoli la varianza di $S$.
````

La variabile casuale $S$ ha la seguente distribuzione di probabilità:

|    $s$     |       2        |       3        |       4        |       5        |       6        |       7        |       8        |       9        |       10       |       11       |       12       |
|:----------:|:--------------:|:--------------:|:--------------:|:--------------:|:--------------:|:--------------:|:--------------:|:--------------:|:--------------:|:--------------:|:--------------:|
| $P(S = s)$ | $\frac{1}{36}$ | $\frac{2}{36}$ | $\frac{3}{36}$ | $\frac{4}{36}$ | $\frac{5}{36}$ | $\frac{6}{36}$ | $\frac{5}{36}$ | $\frac{4}{36}$ | $\frac{3}{36}$ | $\frac{2}{36}$ | $\frac{1}{36}$ |

Essendo $\mathbb{E}(S) = 7$, la varianza diventa

$$
\begin{align}
\mathbb{V}(S) &= \sum \left(s - \mathbb{E}(S)\right)^2 \cdot P(s) \notag\\
&= (2 - 7)^2 \cdot \frac{1}{36} + (3-7)^2 \cdot \frac{3}{36} + \dots + (12 - 7)^2 \cdot \frac{1}{36} \notag\\
&= 5.8333.\notag
\end{align}
$$

Svolgiamo l'esercizio in `R`.

In [27]:
# Definizione dei valori e delle probabilità
x <- 2:12
px <- c(1, 2, 3, 4, 5, 6, 5, 4, 3, 2, 1) / 36

# Calcolo del valore atteso
ex <- sum(x * px)
ex

[1] 7

In [28]:
# Calcolo della varianza
var_x <- sum(((x - ex)^2) * px)
var_x


[1] 5.833333

### Formula alternativa per la varianza

C'è un modo più semplice per calcolare la varianza:

$$
\begin{align}
\mathbb{E}\Big[\big(Y - \mathbb{E}(Y)\big)^2\Big] &= \mathbb{E}\big(Y^2 - 2Y\mathbb{E}(Y) + \mathbb{E}(Y)^2\big)\notag\\
&= \mathbb{E}(Y^2) - 2\mathbb{E}(Y)\mathbb{E}(Y) + \mathbb{E}(Y)^2,
\end{align}
$$

dato che $\mathbb{E}(Y)$ è una costante. Pertanto

$$
\mathbb{V}(Y) = \mathbb{E}(Y^2) - \big(\mathbb{E}(Y) \big)^2.
$$ (eq-def-alt-var-rv)

A parole: la varianza è la media dei quadrati meno il quadrato della media della variabile.


````{prf:example}
:label: rv-expl-7

Consideriamo la variabile casuale $Y$ che corrisponde al numero di teste che si osservano nel lancio di una moneta truccata con probabilità di testa uguale a 0.8. Si trovi la varianza di $Y$.
````

Il valore atteso di $Y$ è

$$
\mathbb{E}(Y) = 0 \cdot 0.2 + 1 \cdot 0.8 = 0.8.
$$

Usando la formula tradizionale della varianza otteniamo:

$$
\mathbb{V}(Y) = (0 - 0.8)^2 \cdot 0.2 + (1 - 0.8)^2 \cdot 0.8 = 0.16.
$$

Lo stesso risultato si trova con la formula alternativa della varianza. Il valore atteso di $Y^2$ è

$$
\mathbb{E}(Y^2) = 0^2 \cdot 0.2 + 1^2 \cdot 0.8 = 0.8.
$$

e la varianza diventa

$$
\mathbb{V}(Y) = \mathbb{E}(Y^2) - \big(\mathbb{E}(Y) \big)^2 = 0.8 - 0.8^2 = 0.16.
$$

Svolgiamo l'esercizio in `R`:

In [29]:
# Definizione dei valori e delle probabilità
y <- c(0, 1)
py <- c(0.2, 0.8)

# Calcolo del valore atteso dei quadrati di y
expected_y_sq <- sum((y^2) * py)

# Calcolo del quadrato del valore atteso di y
sq_expected_y <- (sum(y * py))^2

# Calcolo della varianza
var_y <- expected_y_sq - sq_expected_y

# Stampa del risultato
var_y

[1] 0.16

### Proprietà

**Segno della varianza.** La varianza di una variabile aleatoria non è mai negativa, ed è zero solamente quando la variabile assume  un solo valore.

**Invarianza per traslazione.** La varianza è invariante per traslazione, che lascia fisse le distanze dalla media, e cambia quadraticamente per riscalamento:
  
$$
\mathbb{V}(a + bX) = b^2\mathbb{V}(X).
$$

*Dimostrazione.* Iniziamo a scrivere

$$
(aX+b)-{\mathbb{E}}[aX+b]=aX+b-a{\mathbb{E}}[X]-b=a(X-{\mathbb  {E}}[X]).
$$

Quindi

$$
\sigma _{{aX+b}}^{2}={\mathbb{E}}[a^{2}(X-{\mathbb  {E}}[X])^{2}]=a^{2}\sigma _{X}^{2}.
$$


Esaminiamo una dimostrazione numerica.

In [30]:
# Definizione dei vettori
x <- c(2, 1, 4, 7)
y <- 100 + 2 * x

# Calcolo delle varianze
var_y <- var(y)
var_x <- var(x)

# Verifica dell'uguaglianza
uguaglianza <- var_y == 2^2 * var_x
uguaglianza

[1] TRUE

**Varianza della somma di due variabili indipendenti.** La varianza della somma di due variabili indipendenti o anche solo incorrelate è pari alla somma delle loro varianze:

$$
\mathbb{V}(X+Y) = \mathbb{V}(X) + \mathbb{V}(Y).
$$

*Dimostrazione.* Se $\mathbb{E}(X) = \mathbb{E}(Y) = 0$, allora $\mathbb{E}(X+Y) = 0$ e 

$$\mathbb{V}(X+Y) = \mathbb{E}((X+Y)^2) = \mathbb{E}(X^2) + 2 \mathbb{E}(XY) + \mathbb{E}(Y^2).$$ 

Siccome le variabili sono indipendenti risulta $\mathbb{E}(XY) = \mathbb{E}(X)\mathbb{E}(Y) = 0$. 

**Varianza della differenza di due variabili indipendenti.** La varianza della differenza di due variabili indipendenti è pari alla somma delle loro varianze:

$$
\mathbb{V}(X-Y) = \mathbb{V}(X) + \mathbb{V}(Y).
$$

*Dimostrazione.*

$$
\mathbb{V}(X-Y) = \mathbb{V}(X +(-Y)) = \mathbb{V}(X) + \mathbb{V}(-Y) = \mathbb{V}(X) + \mathbb{V}(Y).
$$

**Varianza della somma di due variabili non indipendenti.** Se $X$ e $Y$ non sono indipendenti, la formula viene corretta dalla loro covarianza:

$$
\mathbb{V}(X+Y) = \mathbb{V}(X) + \mathbb{V}(Y) + 2 Cov(X,Y),
$$

dove $Cov(X,Y) = \mathbb{E}(XY) - \mathbb{E}(X)\mathbb{E}(Y)$.

Una dimostrazione numerica di questo principio è fornita sotto.

In [31]:
# Definizione dei vettori
x <- c(2, 1, 4, 7)
y <- c(1, 3, 5, 11)

# Calcolo della varianza della somma di x e y
var_sum <- var(x + y)

# Calcolo della varianza di x e y separatamente
var_x <- var(x)
var_y <- var(y)

# Calcolo della covarianza tra x e y
cov_xy <- cov(x, y)

# Confronto delle varianze
uguaglianza <- var_sum == var_x + var_y + 2 * cov_xy
uguaglianza


[1] TRUE

**Varianza della media di variabili indipendenti.** La media aritmetica 
$\textstyle {\bar  {X}}={\frac  {X_{1}+\ldots +X_{n}}{n}}$ di $n$ variabili casuali indipendenti aventi la medesima distribuzione, ha varianza 

$$
\mathbb{V}(\bar{X}) = \frac{1}{n^2} \mathbb{V}(X_1)+ \dots \mathbb{V}(X_n) = \frac{1}{n^2} n \mathbb{V}(X) = \frac{1}{n} \mathbb{V}(X).
$$

Il principio precedente è illustrato dalla seguente simulazione.

In [32]:
# Imposta la distribuzione della popolazione
population <- rnorm(n = 10000, mean = 50, sd = 10)

# Imposta la dimensione del campione e il numero di campioni
sample_size <- 30
num_samples <- 100000

# Crea un vettore per tenere traccia delle medie dei campioni
sample_means <- numeric(num_samples)

# Genera i campioni e calcola le loro medie
for (i in 1:num_samples) {
    sample <- sample(population, size = sample_size)
    sample_means[i] <- mean(sample)
}

# Calcola la varianza delle medie campionarie
sampling_dist_mean_var <- var(sample_means)
sampling_dist_mean_var

[1] 3.361443

Il valore teorico della varianza della distribuzione campionaria della media è

In [33]:
10^2 / 30

[1] 3.333333

### Variabili casuali continue

Nel caso di una variabile casuale continua $Y$, la varianza diventa:

$$
\mathbb{V}(Y) = \int_{-\infty}^{+\infty} \large[y - \mathbb{E}(Y)\large]^2 p(y) \,\operatorname {d}\!y.
$$ (eq-def-var-rv-cont)

Come nel caso discreto, la varianza di una v.c. continua $Y$ misura approssimativamente la distanza al quadrato tipica o prevista dei possibili valori $y$ dalla loro media.

## Deviazione standard

Quando lavoriamo con le varianze, i termini sono innalzati al quadrato e quindi i numeri possono diventare molto grandi (o molto piccoli). Per trasformare nuovamente i valori nell'unità di misura della scala originaria si prende la radice quadrata. Il valore risultante viene chiamato *deviazione standard* e solitamente è denotato dalla lettera greca $\sigma$.

```{admonition} Definizione
Si definisce scarto quadratico medio (o deviazione standard o scarto tipo) la radice quadrata della varianza:

$$
\sigma_Y = \sqrt{\mathbb{V}(Y)}.
$$ (eq-def-sd)
```

Come nella statistica descrittiva, la deviazione standard di una variabile casuale misura approssimativamente la distanza tipica o prevista dei possibili valori $y$ dalla loro media.

Per i dadi equilibrati dell'esemio precedebte, la deviazione standard della variabile casuale $S$ è uguale a $\sqrt{5.833} = 2.415$.

## Standardizzazione

```{admonition} Definizione
Data una variabile casuale $Y$, si dice *variabile standardizzata* di $Y$ l'espressione

$$
Z = \frac{Y - \mathbb{E}(Y)}{\sigma_Y}.
$$ (eq-standardization)
```

Solitamente, una variabile standardizzata viene denotata con la lettera $Z$.

## Momenti di variabili casuali

```{admonition} Definizione
Si chiama *momento* di ordine $q$ di una v.c. $X$, dotata di densità $p(x)$, la quantità

$$
\mathbb{E}(X^q) = \int_{-\infty}^{+\infty} x^q p(x) \; dx.
$$ (eq-moments-cont)

Se $X$ è una v.c. discreta, i suoi momenti valgono:

$$
\mathbb{E}(X^q) = \sum_i x_i^q P(x_i).
$$ (eq-moments-discr)
```

I momenti sono importanti parametri indicatori di certe proprietà di $X$. I più noti sono senza dubbio quelli per $q = 1$ e $q = 2$. Il momento del primo ordine corrisponde al valore atteso di $X$. Spesso i momenti di ordine superiore al primo vengono calcolati rispetto al valor medio di $X$, operando una traslazione $x_0 = x − \mathbb{E}(X)$ che individua lo scarto dalla media. Ne deriva che il momento centrale di ordine 2 corrisponde alla varianza.

## Commenti e considerazioni finali 

L'inferenza bayesiana mira a descrivere la distribuzione a posteriori di variabili casuali che rappresentano i parametri di un modello statistico. Nel capitolo precedente, abbiamo esaminato le caratteristiche principali delle variabili casuali, concentrandoci sul caso discreto. In questo capitolo, abbiamo approfondito le proprietà di una singola variabile casuale. Nel prossimo capitolo, invece, esploreremo il problema di descrivere il verificarsi congiunto di due o più variabili casuali.

## Session Info

In [2]:
sessionInfo()

R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] C

time zone: Europe/Rome
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggokabeito_0.1.0  viridis_0.6.5     viridisLite_0.4.2 ggpubr_0.6.0     
 [5] ggExtra_0.10.1    bayesplot_1.11.0  gridExtra_2.3     patchwork_1.2.0  
 [9] semTools_0.5-6    semPlot_1.1.6     lavaan_0.6-17     psych_2.4.1      
[13] scales_1.3.0      markdown_1.12     knitr_1.45        lubridate_1.9.3  
[17] forcats_1.0.0     stringr_1.5.1     dplyr_1.1.4       purrr_1.0.2      
[21] readr_2.1.5       tidyr_1.3.1       tibble_3.2.1      ggplot2_3.4.4    
[25] tidyvers